acceptance: TestCompose failed [KeyError: 'ContainerConfig']

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15459103?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15459103?buildTab=artifacts#/) on master @ [cf7ad52a113613bef9bdf9d380c9eec4d8715b85](https://github.com/cockroachdb/cockroach/commits/cf7ad52a113613bef9bdf9d380c9eec4d8715b85):


```
        OSError: [Errno 30] Read-only file system: '/var/crash/_usr_bin_docker-compose.1001.crash'
        
        Original exception was:
        Traceback (most recent call last):
          File "/bin/docker-compose", line 33, in <module>
            sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 81, in main
            command_func()
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 203, in perform_command
            handler(command, command_options)
          File "/usr/lib/python3/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
            result = fn(*args, **kwargs)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1186, in up
            to_attach = up(False)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1166, in up
            return self.project.up(
          File "/usr/lib/python3/dist-packages/compose/project.py", line 697, in up
            results, errors = parallel.parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/project.py", line 679, in do
            return service.execute_convergence_plan(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 579, in execute_convergence_plan
            return self._execute_convergence_recreate(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
            containers, errors = parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/service.py", line 494, in recreate
            return self.recreate_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 612, in recreate_container
            new_container = self.create_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 330, in create_container
            container_options = self._get_container_create_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 921, in _get_container_create_options
            container_options, override_options = self._build_container_volume_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 960, in _build_container_volume_options
            binds, affinity = merge_volume_bindings(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
            old_volumes, old_mounts = get_container_data_volumes(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
            container.image_config['ContainerConfig'].get('Volumes') or {}
        KeyError: 'ContainerConfig'
        
    compose_test.go:95: exit status 1
    --- FAIL: TestCompose/ComposeGSSPython (20.48s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124862 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1.1-rc release-blocker]
- #124861 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1 release-blocker]
- #124860 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-23.2 release-blocker]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39119

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15454602?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15454602?buildTab=artifacts#/) on master @ [e5f5597b1c1570fff03ee6dfd4a1e8be65c0e008](https://github.com/cockroachdb/cockroach/commits/e5f5597b1c1570fff03ee6dfd4a1e8be65c0e008):


```
        OSError: [Errno 30] Read-only file system: '/var/crash/_usr_bin_docker-compose.1001.crash'
        
        Original exception was:
        Traceback (most recent call last):
          File "/bin/docker-compose", line 33, in <module>
            sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 81, in main
            command_func()
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 203, in perform_command
            handler(command, command_options)
          File "/usr/lib/python3/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
            result = fn(*args, **kwargs)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1186, in up
            to_attach = up(False)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1166, in up
            return self.project.up(
          File "/usr/lib/python3/dist-packages/compose/project.py", line 697, in up
            results, errors = parallel.parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/project.py", line 679, in do
            return service.execute_convergence_plan(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 579, in execute_convergence_plan
            return self._execute_convergence_recreate(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
            containers, errors = parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/service.py", line 494, in recreate
            return self.recreate_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 612, in recreate_container
            new_container = self.create_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 330, in create_container
            container_options = self._get_container_create_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 921, in _get_container_create_options
            container_options, override_options = self._build_container_volume_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 960, in _build_container_volume_options
            binds, affinity = merge_volume_bindings(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
            old_volumes, old_mounts = get_container_data_volumes(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
            container.image_config['ContainerConfig'].get('Volumes') or {}
        KeyError: 'ContainerConfig'
        
    compose_test.go:95: exit status 1
    --- FAIL: TestCompose/ComposeGSSPython (19.12s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124862 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1.1-rc release-blocker]
- #124861 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1 release-blocker]
- #124860 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15460913?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15460913?buildTab=artifacts#/) on master @ [e4d582940d4aaf48e3605898fd4f694750d91767](https://github.com/cockroachdb/cockroach/commits/e4d582940d4aaf48e3605898fd4f694750d91767):


```
        OSError: [Errno 30] Read-only file system: '/var/crash/_usr_bin_docker-compose.1001.crash'
        
        Original exception was:
        Traceback (most recent call last):
          File "/bin/docker-compose", line 33, in <module>
            sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 81, in main
            command_func()
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 203, in perform_command
            handler(command, command_options)
          File "/usr/lib/python3/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
            result = fn(*args, **kwargs)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1186, in up
            to_attach = up(False)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1166, in up
            return self.project.up(
          File "/usr/lib/python3/dist-packages/compose/project.py", line 697, in up
            results, errors = parallel.parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/project.py", line 679, in do
            return service.execute_convergence_plan(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 579, in execute_convergence_plan
            return self._execute_convergence_recreate(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
            containers, errors = parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/service.py", line 494, in recreate
            return self.recreate_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 612, in recreate_container
            new_container = self.create_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 330, in create_container
            container_options = self._get_container_create_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 921, in _get_container_create_options
            container_options, override_options = self._build_container_volume_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 960, in _build_container_volume_options
            binds, affinity = merge_volume_bindings(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
            old_volumes, old_mounts = get_container_data_volumes(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
            container.image_config['ContainerConfig'].get('Volumes') or {}
        KeyError: 'ContainerConfig'
        
    compose_test.go:95: exit status 1
    --- FAIL: TestCompose/ComposeGSSPython (20.10s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124862 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1.1-rc]
- #124861 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1]
- #124860 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15461006?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15461006?buildTab=artifacts#/) on master @ [e0ac711692c414295a6075038e0a5862435118f8](https://github.com/cockroachdb/cockroach/commits/e0ac711692c414295a6075038e0a5862435118f8):


```
        OSError: [Errno 30] Read-only file system: '/var/crash/_usr_bin_docker-compose.1001.crash'
        
        Original exception was:
        Traceback (most recent call last):
          File "/bin/docker-compose", line 33, in <module>
            sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 81, in main
            command_func()
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 203, in perform_command
            handler(command, command_options)
          File "/usr/lib/python3/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
            result = fn(*args, **kwargs)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1186, in up
            to_attach = up(False)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1166, in up
            return self.project.up(
          File "/usr/lib/python3/dist-packages/compose/project.py", line 697, in up
            results, errors = parallel.parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/project.py", line 679, in do
            return service.execute_convergence_plan(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 579, in execute_convergence_plan
            return self._execute_convergence_recreate(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
            containers, errors = parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/service.py", line 494, in recreate
            return self.recreate_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 612, in recreate_container
            new_container = self.create_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 330, in create_container
            container_options = self._get_container_create_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 921, in _get_container_create_options
            container_options, override_options = self._build_container_volume_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 960, in _build_container_volume_options
            binds, affinity = merge_volume_bindings(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
            old_volumes, old_mounts = get_container_data_volumes(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
            container.image_config['ContainerConfig'].get('Volumes') or {}
        KeyError: 'ContainerConfig'
        
    compose_test.go:95: exit status 1
    --- FAIL: TestCompose/ComposeGSSPython (20.25s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124862 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1.1-rc]
- #124861 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1]
- #124860 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15461881?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15461881?buildTab=artifacts#/) on master @ [7a9fdab9bd0f3e6e39be4e7075922aa2cdfb6300](https://github.com/cockroachdb/cockroach/commits/7a9fdab9bd0f3e6e39be4e7075922aa2cdfb6300):


```
        OSError: [Errno 30] Read-only file system: '/var/crash/_usr_bin_docker-compose.1001.crash'
        
        Original exception was:
        Traceback (most recent call last):
          File "/bin/docker-compose", line 33, in <module>
            sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 81, in main
            command_func()
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 203, in perform_command
            handler(command, command_options)
          File "/usr/lib/python3/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
            result = fn(*args, **kwargs)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1186, in up
            to_attach = up(False)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1166, in up
            return self.project.up(
          File "/usr/lib/python3/dist-packages/compose/project.py", line 697, in up
            results, errors = parallel.parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/project.py", line 679, in do
            return service.execute_convergence_plan(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 579, in execute_convergence_plan
            return self._execute_convergence_recreate(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
            containers, errors = parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/service.py", line 494, in recreate
            return self.recreate_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 612, in recreate_container
            new_container = self.create_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 330, in create_container
            container_options = self._get_container_create_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 921, in _get_container_create_options
            container_options, override_options = self._build_container_volume_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 960, in _build_container_volume_options
            binds, affinity = merge_volume_bindings(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
            old_volumes, old_mounts = get_container_data_volumes(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
            container.image_config['ContainerConfig'].get('Volumes') or {}
        KeyError: 'ContainerConfig'
        
    compose_test.go:95: exit status 1
    --- FAIL: TestCompose/ComposeGSSPython (19.51s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124862 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1.1-rc]
- #124861 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1]
- #124860 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15462823?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15462823?buildTab=artifacts#/) on master @ [04ecf34d21b0a49c57753096d2926121a8356689](https://github.com/cockroachdb/cockroach/commits/04ecf34d21b0a49c57753096d2926121a8356689):


```
        OSError: [Errno 30] Read-only file system: '/var/crash/_usr_bin_docker-compose.1001.crash'
        
        Original exception was:
        Traceback (most recent call last):
          File "/bin/docker-compose", line 33, in <module>
            sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 81, in main
            command_func()
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 203, in perform_command
            handler(command, command_options)
          File "/usr/lib/python3/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
            result = fn(*args, **kwargs)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1186, in up
            to_attach = up(False)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1166, in up
            return self.project.up(
          File "/usr/lib/python3/dist-packages/compose/project.py", line 697, in up
            results, errors = parallel.parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/project.py", line 679, in do
            return service.execute_convergence_plan(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 579, in execute_convergence_plan
            return self._execute_convergence_recreate(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
            containers, errors = parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/service.py", line 494, in recreate
            return self.recreate_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 612, in recreate_container
            new_container = self.create_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 330, in create_container
            container_options = self._get_container_create_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 921, in _get_container_create_options
            container_options, override_options = self._build_container_volume_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 960, in _build_container_volume_options
            binds, affinity = merge_volume_bindings(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
            old_volumes, old_mounts = get_container_data_volumes(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
            container.image_config['ContainerConfig'].get('Volumes') or {}
        KeyError: 'ContainerConfig'
        
    compose_test.go:95: exit status 1
    --- FAIL: TestCompose/ComposeGSSPython (20.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124862 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1.1-rc]
- #124861 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1]
- #124860 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15463890?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15463890?buildTab=artifacts#/) on master @ [d8e609f02a3d19bc82bca71df8fd6642113763ee](https://github.com/cockroachdb/cockroach/commits/d8e609f02a3d19bc82bca71df8fd6642113763ee):


```
        OSError: [Errno 30] Read-only file system: '/var/crash/_usr_bin_docker-compose.1001.crash'
        
        Original exception was:
        Traceback (most recent call last):
          File "/bin/docker-compose", line 33, in <module>
            sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 81, in main
            command_func()
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 203, in perform_command
            handler(command, command_options)
          File "/usr/lib/python3/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
            result = fn(*args, **kwargs)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1186, in up
            to_attach = up(False)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1166, in up
            return self.project.up(
          File "/usr/lib/python3/dist-packages/compose/project.py", line 697, in up
            results, errors = parallel.parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/project.py", line 679, in do
            return service.execute_convergence_plan(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 579, in execute_convergence_plan
            return self._execute_convergence_recreate(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
            containers, errors = parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/service.py", line 494, in recreate
            return self.recreate_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 612, in recreate_container
            new_container = self.create_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 330, in create_container
            container_options = self._get_container_create_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 921, in _get_container_create_options
            container_options, override_options = self._build_container_volume_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 960, in _build_container_volume_options
            binds, affinity = merge_volume_bindings(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
            old_volumes, old_mounts = get_container_data_volumes(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
            container.image_config['ContainerConfig'].get('Volumes') or {}
        KeyError: 'ContainerConfig'
        
    compose_test.go:95: exit status 1
    --- FAIL: TestCompose/ComposeGSSPython (17.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124883 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-23.2.6-rc]
- #124862 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1.1-rc]
- #124861 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1]
- #124860 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15464408?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15464408?buildTab=artifacts#/) on master @ [c5168acee12aba99154be04d58013d5e8defd22c](https://github.com/cockroachdb/cockroach/commits/c5168acee12aba99154be04d58013d5e8defd22c):


```
        OSError: [Errno 30] Read-only file system: '/var/crash/_usr_bin_docker-compose.1001.crash'
        
        Original exception was:
        Traceback (most recent call last):
          File "/bin/docker-compose", line 33, in <module>
            sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 81, in main
            command_func()
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 203, in perform_command
            handler(command, command_options)
          File "/usr/lib/python3/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
            result = fn(*args, **kwargs)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1186, in up
            to_attach = up(False)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1166, in up
            return self.project.up(
          File "/usr/lib/python3/dist-packages/compose/project.py", line 697, in up
            results, errors = parallel.parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/project.py", line 679, in do
            return service.execute_convergence_plan(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 579, in execute_convergence_plan
            return self._execute_convergence_recreate(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
            containers, errors = parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/service.py", line 494, in recreate
            return self.recreate_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 612, in recreate_container
            new_container = self.create_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 330, in create_container
            container_options = self._get_container_create_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 921, in _get_container_create_options
            container_options, override_options = self._build_container_volume_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 960, in _build_container_volume_options
            binds, affinity = merge_volume_bindings(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
            old_volumes, old_mounts = get_container_data_volumes(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
            container.image_config['ContainerConfig'].get('Volumes') or {}
        KeyError: 'ContainerConfig'
        
    compose_test.go:95: exit status 1
    --- FAIL: TestCompose/ComposeGSSPython (17.38s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124883 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-23.2.6-rc]
- #124862 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1.1-rc]
- #124861 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1]
- #124860 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15464652?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15464652?buildTab=artifacts#/) on master @ [2bf8e36e00f920fe05bdcf26af5b7f464ec21e62](https://github.com/cockroachdb/cockroach/commits/2bf8e36e00f920fe05bdcf26af5b7f464ec21e62):


```
        OSError: [Errno 30] Read-only file system: '/var/crash/_usr_bin_docker-compose.1001.crash'
        
        Original exception was:
        Traceback (most recent call last):
          File "/bin/docker-compose", line 33, in <module>
            sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 81, in main
            command_func()
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 203, in perform_command
            handler(command, command_options)
          File "/usr/lib/python3/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
            result = fn(*args, **kwargs)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1186, in up
            to_attach = up(False)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1166, in up
            return self.project.up(
          File "/usr/lib/python3/dist-packages/compose/project.py", line 697, in up
            results, errors = parallel.parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/project.py", line 679, in do
            return service.execute_convergence_plan(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 579, in execute_convergence_plan
            return self._execute_convergence_recreate(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
            containers, errors = parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/service.py", line 494, in recreate
            return self.recreate_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 612, in recreate_container
            new_container = self.create_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 330, in create_container
            container_options = self._get_container_create_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 921, in _get_container_create_options
            container_options, override_options = self._build_container_volume_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 960, in _build_container_volume_options
            binds, affinity = merge_volume_bindings(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
            old_volumes, old_mounts = get_container_data_volumes(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
            container.image_config['ContainerConfig'].get('Volumes') or {}
        KeyError: 'ContainerConfig'
        
    compose_test.go:95: exit status 1
    --- FAIL: TestCompose/ComposeGSSPython (16.95s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124883 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-23.2.6-rc]
- #124862 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1.1-rc]
- #124861 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1]
- #124860 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15465052?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15465052?buildTab=artifacts#/) on master @ [64bd4bbac8f8d27a2fe4968404cc3d048480375f](https://github.com/cockroachdb/cockroach/commits/64bd4bbac8f8d27a2fe4968404cc3d048480375f):


```
        OSError: [Errno 30] Read-only file system: '/var/crash/_usr_bin_docker-compose.1001.crash'
        
        Original exception was:
        Traceback (most recent call last):
          File "/bin/docker-compose", line 33, in <module>
            sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 81, in main
            command_func()
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 203, in perform_command
            handler(command, command_options)
          File "/usr/lib/python3/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
            result = fn(*args, **kwargs)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1186, in up
            to_attach = up(False)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1166, in up
            return self.project.up(
          File "/usr/lib/python3/dist-packages/compose/project.py", line 697, in up
            results, errors = parallel.parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/project.py", line 679, in do
            return service.execute_convergence_plan(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 579, in execute_convergence_plan
            return self._execute_convergence_recreate(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
            containers, errors = parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/service.py", line 494, in recreate
            return self.recreate_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 612, in recreate_container
            new_container = self.create_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 330, in create_container
            container_options = self._get_container_create_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 921, in _get_container_create_options
            container_options, override_options = self._build_container_volume_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 960, in _build_container_volume_options
            binds, affinity = merge_volume_bindings(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
            old_volumes, old_mounts = get_container_data_volumes(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
            container.image_config['ContainerConfig'].get('Volumes') or {}
        KeyError: 'ContainerConfig'
        
    compose_test.go:95: exit status 1
    --- FAIL: TestCompose/ComposeGSSPython (17.99s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124883 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-23.2.6-rc]
- #124862 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1.1-rc]
- #124861 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-24.1]
- #124860 acceptance: TestCompose failed [C-test-failure O-robot T-sql-foundations branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15466427?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15466427?buildTab=artifacts#/) on master @ [64bd4bbac8f8d27a2fe4968404cc3d048480375f](https://github.com/cockroachdb/cockroach/commits/64bd4bbac8f8d27a2fe4968404cc3d048480375f):


```
        OSError: [Errno 30] Read-only file system: '/var/crash/_usr_bin_docker-compose.1001.crash'
        
        Original exception was:
        Traceback (most recent call last):
          File "/bin/docker-compose", line 33, in <module>
            sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 81, in main
            command_func()
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 203, in perform_command
            handler(command, command_options)
          File "/usr/lib/python3/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
            result = fn(*args, **kwargs)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1186, in up
            to_attach = up(False)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1166, in up
            return self.project.up(
          File "/usr/lib/python3/dist-packages/compose/project.py", line 697, in up
            results, errors = parallel.parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/project.py", line 679, in do
            return service.execute_convergence_plan(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 579, in execute_convergence_plan
            return self._execute_convergence_recreate(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
            containers, errors = parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/service.py", line 494, in recreate
            return self.recreate_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 612, in recreate_container
            new_container = self.create_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 330, in create_container
            container_options = self._get_container_create_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 921, in _get_container_create_options
            container_options, override_options = self._build_container_volume_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 960, in _build_container_volume_options
            binds, affinity = merge_volume_bindings(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
            old_volumes, old_mounts = get_container_data_volumes(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
            container.image_config['ContainerConfig'].get('Volumes') or {}
        KeyError: 'ContainerConfig'
        
    compose_test.go:95: exit status 1
    --- FAIL: TestCompose/ComposeGSSPython (16.74s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124883 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-23.2.6-rc]
- #124862 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-24.1.1-rc]
- #124861 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-24.1]
- #124860 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15470297?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15470297?buildTab=artifacts#/) on master @ [962b22a5c34989212c0638ca671b5f9949c95a9f](https://github.com/cockroachdb/cockroach/commits/962b22a5c34989212c0638ca671b5f9949c95a9f):


```
        OSError: [Errno 30] Read-only file system: '/var/crash/_usr_bin_docker-compose.1001.crash'
        
        Original exception was:
        Traceback (most recent call last):
          File "/bin/docker-compose", line 33, in <module>
            sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 81, in main
            command_func()
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 203, in perform_command
            handler(command, command_options)
          File "/usr/lib/python3/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
            result = fn(*args, **kwargs)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1186, in up
            to_attach = up(False)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1166, in up
            return self.project.up(
          File "/usr/lib/python3/dist-packages/compose/project.py", line 697, in up
            results, errors = parallel.parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/project.py", line 679, in do
            return service.execute_convergence_plan(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 579, in execute_convergence_plan
            return self._execute_convergence_recreate(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
            containers, errors = parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/service.py", line 494, in recreate
            return self.recreate_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 612, in recreate_container
            new_container = self.create_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 330, in create_container
            container_options = self._get_container_create_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 921, in _get_container_create_options
            container_options, override_options = self._build_container_volume_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 960, in _build_container_volume_options
            binds, affinity = merge_volume_bindings(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
            old_volumes, old_mounts = get_container_data_volumes(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
            container.image_config['ContainerConfig'].get('Volumes') or {}
        KeyError: 'ContainerConfig'
        
    compose_test.go:95: exit status 1
    --- FAIL: TestCompose/ComposeGSSPython (0.30s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124883 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-23.2.6-rc]
- #124862 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-24.1.1-rc]
- #124861 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-24.1]
- #124860 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15470561?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15470561?buildTab=artifacts#/) on master @ [d523f75f7d7006d45957f78c0381dd9774e78713](https://github.com/cockroachdb/cockroach/commits/d523f75f7d7006d45957f78c0381dd9774e78713):


```
        OSError: [Errno 30] Read-only file system: '/var/crash/_usr_bin_docker-compose.1001.crash'
        
        Original exception was:
        Traceback (most recent call last):
          File "/bin/docker-compose", line 33, in <module>
            sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 81, in main
            command_func()
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 203, in perform_command
            handler(command, command_options)
          File "/usr/lib/python3/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
            result = fn(*args, **kwargs)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1186, in up
            to_attach = up(False)
          File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1166, in up
            return self.project.up(
          File "/usr/lib/python3/dist-packages/compose/project.py", line 697, in up
            results, errors = parallel.parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/project.py", line 679, in do
            return service.execute_convergence_plan(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 579, in execute_convergence_plan
            return self._execute_convergence_recreate(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
            containers, errors = parallel_execute(
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
            raise error_to_reraise
          File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
            result = func(obj)
          File "/usr/lib/python3/dist-packages/compose/service.py", line 494, in recreate
            return self.recreate_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 612, in recreate_container
            new_container = self.create_container(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 330, in create_container
            container_options = self._get_container_create_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 921, in _get_container_create_options
            container_options, override_options = self._build_container_volume_options(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 960, in _build_container_volume_options
            binds, affinity = merge_volume_bindings(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
            old_volumes, old_mounts = get_container_data_volumes(
          File "/usr/lib/python3/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
            container.image_config['ContainerConfig'].get('Volumes') or {}
        KeyError: 'ContainerConfig'
        
    compose_test.go:95: exit status 1
    --- FAIL: TestCompose/ComposeGSSPython (17.85s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124883 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-23.2.6-rc]
- #124862 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-24.1.1-rc]
- #124861 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-24.1]
- #124860 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/acceptance/compose/flyway/docker-compose.yml
+++ b/pkg/acceptance/compose/flyway/docker-compose.yml
@@ -1,4 +1,3 @@
-version: '3'
 services:
   cockroach:
     # We use a docker image mirror to avoid pulling from 3rd party repos, which sometimes have reliability issues.

--- a/pkg/acceptance/compose/gss/docker-compose-python.yml
+++ b/pkg/acceptance/compose/gss/docker-compose-python.yml
@@ -1,4 +1,3 @@
-version: '3'
 services:
   kdc:
     image: us-east1-docker.pkg.dev/crl-ci-images/cockroach/acceptance-gss-kdc:20221214-131000

--- a/pkg/acceptance/compose/gss/docker-compose.yml
+++ b/pkg/acceptance/compose/gss/docker-compose.yml
@@ -1,4 +1,3 @@
-version: '3'
 services:
   kdc:
     image: us-east1-docker.pkg.dev/crl-ci-images/cockroach/acceptance-gss-kdc:20221214-131000

--- a/pkg/acceptance/compose_test.go
+++ b/pkg/acceptance/compose_test.go
@@ -74,8 +74,13 @@ func testCompose(t *testing.T, path string, exitCodeFrom string) {
 		t.Fatalf(err.Error())
 	}
 	cmd := exec.Command(
-		"docker-compose",
-		"--no-ansi",
+		"docker",
+		"compose",
+		// NB: Using --compatibility here in order to preserve compose V1 hostnames
+		// (with underscores) instead of V2 hostnames (with -), because the
+		// hostnames are hardcoded in the Kerberos keys.
+		"--compatibility",
+		"--ansi=never",
 		"-f", path,
 		"up",
 		"--force-recreate",

--- a/pkg/compose/compare/docker-compose.yml
+++ b/pkg/compose/compare/docker-compose.yml
@@ -1,4 +1,3 @@
-version: '3'
 services:
   cockroach1:
     # We use a docker image mirror to avoid pulling from 3rd party repos, which sometimes have reliability issues.

--- a/pkg/compose/compose_test.go
+++ b/pkg/compose/compose_test.go
@@ -121,7 +121,7 @@ func TestComposeCompare(t *testing.T) {
 		"docker",
 		"compose",
 		"-f", dockerComposeYml,
-		"--no-ansi",
+		"--ansi=never",
 		"up",
 		"--force-recreate",
 		"--exit-code-from", "test",

